packages/libs/react-ui/src/components/NavAccordion/NavAccordionSection.tsx

```diff
@@ -0,0 +1,86 @@
+'use client';
+
+import { NavAccordionContext } from './NavAccordion.context';
+import {
+  navAccordionListClass,
+  navAccordionListItemClass,
+} from './NavAccordion.css';
+import type { INavAccordionGroupProps } from './NavAccordionGroup';
+import { NavAccordionGroup } from './NavAccordionGroup';
+import type { INavAccordionLinkProps } from './NavAccordionLink';
+
+import {
+  accordionButtonClass,
+  accordionHeadingTitleClass,
+  accordionSectionClass,
+  accordionToggleIconClass,
+} from '@components/Accordion/Accordion.css';
+import { SystemIcon } from '@components/Icon';
+import classNames from 'classnames';
+import type { FC, FunctionComponentElement as FCElement } from 'react';
+import React, { Children, useContext } from 'react';
+
+export interface INavAccordionSectionProps {
+  children?:
+    | FCElement<INavAccordionGroupProps>[]
+    | FCElement<INavAccordionLinkProps>[]
+    | FCElement<INavAccordionGroupProps>
+    | FCElement<INavAccordionLinkProps>;
+  onClose?: () => void;
+  onOpen?: () => void;
+  title: string;
+}
+
+export const NavAccordionSection: FC<INavAccordionSectionProps> = ({
+  children,
+  onClose,
+  onOpen,
+  title,
+}) => {
+  const { openSections, setOpenSections, linked } =
+    useContext(NavAccordionContext);
+  const sectionId = title.replace(/\s+/g, '').toLowerCase();
+  const isOpen = openSections.includes(sectionId);
+
+  const handleClick = (): void => {
+    if (isOpen) {
+      setOpenSections(
+        linked ? [] : [...openSections.filter((i) => i !== sectionId)],
+      );
+      onClose?.();
+    } else {
+      setOpenSections(linked ? [sectionId] : [...openSections, sectionId]);
+      onOpen?.();
+    }
+  };
+
+  return (
+    <section className={accordionSectionClass}>
```

Comment:
nit: Looking into accessibility, I'm not sure this is the right element to use here. I think a div might make more sense, but I don't have a strong preference: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/section

Reply:
I added a comment to this ticket: https://app.asana.com/0/1205693948735844/1205687963086449/f. I think when we pick that up it would be a good one to also check this and change it if necessary. What do you think?